qiskit/optimization/problems/objective.py

```diff
@@ -174,24 +173,20 @@ def _set(i, j, val):
                     raise QiskitOptimizationError(
                         "set_quadratic expects a list of the length equal to the number of "
                         "variables, where each entry has a pair of the indices of the other "
-                        "variables and values, or the corresponding SparsePair")
+                        "variables and values, or the corresponding SparsePair"
+                        "{}".format(args)
+                    )
 
     def set_quadratic_coefficients(self, *args):
```

Comment:
Because CPLEX API accepts various arguments, the type is super complicated. That's why there are many `if` statements related `args`. I also want to add type hint, but I gave up some cases.

Reply:
In this method, the type of the argument is `List[Union[float, SparsePair, Sequence[Sequence, Sequence]]]`. `Sequence[Sequence, Sequence]]` is tricky because the outer `Sequence` should have size 2 and two internal `Sequence`s inside should have the same size... This is CPLEX design...

Reply:
Most arguments have similar complexity.

Reply:
I agree it is a bit cumbersome. If it is too much work I think it is okay to just put a detailed documentation in the docstring and leave the type hint out.

What do you mean exactly by `List[Sequence[Union[int, float]]]]`? 
Could you give an example? Usually if the Sequence has a fixed length we can use `Tuple`, e.g. `Tuple[int, float]` for types such as `(1, 0.1)`.

Reply:
This is an example. It corresponds to `List[SparsePair]`.
```[
  [[0, 1, 2], [1.0, 2.0, 3.0]],
  ([0, 3, 4], [1.0, -1.0, 2.0])
]```

Reply:
Sorry. `List[Sequence[Union[int, float]]]]` might be wrong.
`List[Sequence[List[Union[int, float]]]]` might be correct.

Reply:
I think the way you put it know is fine. Reading the third paragraph of the comments users can understand what is meant by `List[Sequence]`.

Reply:
Yes, I revised the last items as `List[Sequence]`